src/meta.rs

```diff
@@ -2,21 +2,78 @@ use crate::id::ResourceId;
 
 use anyhow::Error;
 use canonical_path::CanonicalPathBuf;
+use serde::{Deserialize, Serialize};
+use std::any;
 use std::ffi::{OsStr, OsString};
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
 use std::time::SystemTime;
 use walkdir::DirEntry;
 
-#[derive(Eq, PartialEq, Hash, Clone, Debug)]
+const METADATA_RELATIVE_PATH: &str = ".ark/meta";
+
+#[derive(Eq, PartialEq, Hash, Clone, Debug, Deserialize, Serialize)]
 pub struct ResourceMeta {
     pub id: ResourceId,
-    pub modified: SystemTime,
+    // pub modified: SystemTime,
+    pub created_date: SystemTime,
     pub name: Option<OsString>,
     pub extension: Option<OsString>,
     pub kind: Option<ResourceKind>,
     pub extra: Option<ResourceExtra>,
 }
 
 impl ResourceMeta {
+    pub fn store<P: AsRef<Path>>(
+        root: P,
+        path: P,
+        resource_id: ResourceId,
+        name: Option<OsString>,
+        extension: Option<OsString>,
+        kind: Option<ResourceKind>,
+        extra: Option<ResourceExtra>,
+    ) {
+        let metadata_path = root.as_ref().join(METADATA_RELATIVE_PATH);
+        fs::create_dir_all(metadata_path.to_owned())
+            .expect(&format!("Creating {} directory", METADATA_RELATIVE_PATH));
+        let mut metadata_file = File::create(
+            metadata_path
+                .to_owned()
+                .join(format!("{}", resource_id.crc32)),
+        )
+        .unwrap();
+        let metadata = Self {
+            id: resource_id,
+            created_date: SystemTime::now(),
+            name,
+            extension,
+            kind,
+            extra,
```

Comment:
Actually, we only need to store `extra` part. We can put `created_data` into `extra` structure. It's not needed to store `id` (because it is stored in the filename), `name` and `extension` (because it's derived from path which is stored in index). I'm not sure about `kind`, if we can avoid storing it, that would be good. We should store only strictly necessary  data.

Reply:
I think maybe we should keep the id field as it's needed to pass the index tests!